### PR TITLE
Wikimania hackthon work 

### PR DESCRIPTION
**Pull request by dattaz**
Improuve perf and speed and close issue : #30 #27 #25 #23 #22 #21 #20 #19 #17 #9 
and maybe #15

### CODE REVIEW

**Comment by amirouche** [comment]
Thanks I'll check the commits and merge asap!

**Comment by kelson42** [comment]
Hope to be able to test this new versino of Sotoki this WE. Happy review @amirouche

File: sotoki.py [hunk 1]
```diff
@@ -328,19 +328,19 @@ def offline(output, cores):
...
     conn = sqlite3.connect(db)
     conn.row_factory = dict_factory
     cursor = conn.cursor()
     #create table tags-questions
-#    
     cursor.execute("""CREATE TABLE IF NOT EXISTS questiontag(id INTEGER PRIMARY KEY AUTOINCREMENT UNIQUE, Score INTEGER, Title TEXT, CreationDate TEXT, Tag TEXT )""")
     conn.commit()
     questions = cursor.execute("""SELECT * FROM posts WHERE PostTypeId == 1""").fetchall()
     os.makedirs(os.path.join(output, 'question'))
+    all_q = [ ]
```
Comment (amirouche) [inline sotoki.py]: please avoid shortening variable names in the future

Comment (amirouche) [inline sotoki.py]: I prefer `questions` to `all_q`

**Comment by amirouche** [comment]
@dattaz Please use a linter to avoid pep8 and indentation mistakes

**Pull request merged**